src/query/storage/fanout/storage.go

```diff
@@ -115,7 +115,7 @@ func (s *fanoutStorage) FetchBlocks(
 
 			if err != nil {
 				if warning, err := storage.IsWarning(store, err); warning {
-					resultMeta.AddWarning(store.Name(), err.Error())
+					resultMeta.AddWarning(store.Name(), "fetch_blocks_warning")
```

Comment:
Hm, do we lose the actual error with this? 

I suppose we could also do something like this:
```go
resultMeta.AddWarning(fmt.Sprintf("%s_fetch_blocks_warning", store.Name()), err.Error())
```

Or do we actually do an exact string match on `fetch_blocks_warning` somewhere else? (if so probably should be using a constant here).

Anyway would be interesting to hear more of the context on this one either way (looking forward to being on the same continent again soon hah).

Reply:
So we do, but we only lose it in the warning; I was adding the full error to warning headers initially but it leads to pretty gross looking output (especially with the [[totally arbitrary]] output format I'm currently using; figure that at the moment if you receive a warning you can check the query logs for more info?

Thoughts?

Reply:
Yeah that makes sense, perhaps we can make it just a little more user friendly like: "unable to fetch results: see logs for more details."?

At the end of the day, if we actually want to present to the user we'll need to format it in a way that Grafana/whatever frontend application is rendering the data can parse the data.

And if that's the case, we may want to think about returning these as JSON or something, even though it seems kind of weird to do so in a header (lol).

Reply:
I'll put up another PR soon to JSONify these headers